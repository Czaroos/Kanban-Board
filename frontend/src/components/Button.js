import React from "react";
import styled from "styled-components";
import ButtonOld from "@material-ui/core/Button";

const StyledButton = styled(ButtonOld)`
&& {
    color: white;
    background: #5aac44;
}
`;

const Button = ({ children, onClick }) => {
    return (
        <StyledButton variant="contained"
        onMouseDown={onClick}>{children}</StyledButton>
    );
};

export default Button;