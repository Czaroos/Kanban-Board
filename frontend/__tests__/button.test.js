import React from 'react';
import ReactDOM from 'react-dom'
import Button from './../src/components/Button'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'
afterEach(cleanup);

it("render without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>,div)
})

it("render button correctly with text content", ()=>{
  const {getByTestId} = render(<Button children="Add column"></Button>)
  expect(getByTestId('button')).toHaveTextContent("Add column")
})

it("render button correctly with empty text content", ()=>{
    const {getByTestId} = render(<Button children=""></Button>)
    expect(getByTestId('button')).toHaveTextContent("")
  })

it("matches snapshot normal text", ()=>{
    const tree = renderer.create(<Button children="Add task"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
}) 

it("matches snapshot long number", ()=>{
    const tree = renderer.create(<Button children="9999999"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot wierd text", ()=>{
    const tree = renderer.create(<Button children="chomik dÅ¼ungarski"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})