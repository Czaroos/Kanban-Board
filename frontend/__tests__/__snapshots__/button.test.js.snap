// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot long number 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AxjAm hudJyp"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    9999999
  </span>
</button>
`;

exports[`matches snapshot normal text 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AxjAm hudJyp"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Add task
  </span>
</button>
`;

exports[`matches snapshot wierd text 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AxjAm hudJyp"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    chomik d≈ºungarski
  </span>
</button>
`;
